packages/graphql_codegen/test/assets/fragments_and_loose_type_requirements/document.graphql.dart

```diff
@@ -71,6 +71,32 @@ extension UtilityExtension$Fragment$F on Fragment$F {
         this,
         (i) => i,
       );
+  _T when<_T>({required _T Function(Fragment$F$$T) t}) {
+    switch ($__typename) {
```

Comment:
What do you think about making this a type check instead of checking the `__typename` property? Then we can remove an unnecessary cast.

Reply:
Settled on using typename per top-level discussion